File: bench/io/bench_slurp.js
// Read the whole file into memory. See bench_slurp.c for what this measures.
import * as std from "std";
const t0 = Date.now();
const data = std.loadFile("io_input.txt");
const ms = Date.now() - t0;
console.log("result: " + data.length);
console.log("elapsed: " + ms);

File: bench/io/bench_slurp.go
// Read the whole file into memory. See bench_slurp.c for what this measures.
package main

import (
	"fmt"
	"os"
	"time"
)

func main() {
	t0 := time.Now()
	data, err := os.ReadFile("io_input.txt")
	if err != nil {
		fmt.Fprintln(os.Stderr, err)
		os.Exit(1)
	}
	ms := time.Since(t0).Milliseconds()
	fmt.Printf("result: %d\n", len(data))
	fmt.Printf("elapsed: %d\n", ms)
}
